### ISSUE TITLE
Write a simple test

### ISSUE THREAD

**Issue by zmwangx** [zmwangx/storyboard#2]
The test can go like follows: use `tempfile.mkdtemp` to create a temporary working space, then in that directory do the following (of course, need to be translated to python):

``` bash
#!/usr/bin/env bash
colors=(red orange yellow green blue indigo violet)
for color in ${colors[@]}; do
    ffmpeg -f lavfi -i color=c=${color}:s=320x180:d=1 ${color}.mp4;
done
for i in {0..15}; do
    echo file \'${colors[$((i % 7))]}.mp4\' >>list
done
ffmpeg -f concat -i list -c copy rainbow.mp4
storyboard rainbow.mp4
```

The generated storyboard should look something like

![storyboard.jpg](http://i.imgur.com/KTy8Miv.jpg)

**Comment by zmwangx** [comment]
Also, the first thing the test should do is of course to test the existence of `ffmpeg` and `ffprobe` (or there `.exe` cousins.

**Comment by zmwangx** [comment]
By the way, I should also change my directory structure to

```
/root
    /src
        /storyboard
    /tests
```

**Comment by zmwangx** [comment]
Blocked by #1 .

**Comment by zmwangx** [comment]
I might as well just use a simpler version

```
ffmpeg -f lavfi -i color=c=pink:s=320x180:d=10 tempfile.mp4
```

and write a unit test around it.

`metadata` output on this should be

```
Filename:               tempfile.mp4
File size:              8511 (8.32KiB)
Container format:       MPEG-4 Part 14 (MP4)
Duration:               00:00:10.00
Pixel dimensions:       320x180
Display aspect ratio:   16:9
Scan type:              Progressive scan
Frame rate:             25 fps
Streams:
    #0: Video, H.264 (High Profile level 1.2), 320x180 (DAR 16:9), 25 fps, 4 kb/s
```

(Tested on both OS X with FFmpeg 2.6.1 and Ubuntu with FFmpeg 2.6.2. Both with `--enable-libx264`. I'll just assume this output rather than painstakingly figure out a regex that would maximally match all possible configs of FFmpeg. I expect FFmpeg to be stable on something this simple. Even it fails it shouldn't be a big problem, since we can easily see the difference in unit test report; after all this is just one unit test.)

**Comment by zmwangx** [comment]
Unit tests implemented in 50ade87576e091ddbfc26185b8b6925d8d674490 and 10283cdde0181365e35018b6734173e8943b3685.

**Comment by zmwangx** [comment]
Turns out that the `metadata` output above isn't bullet-proof. Some other version on Ubuntu returned  a file with 8512 bytes:

```
Filename:               tempfile.mp4
File size:              8512 (8.32KiB)
Container format:       MPEG-4 Part 14 (MP4)
Duration:               00:00:10.00
Pixel dimensions:       320x180
Display aspect ratio:   16:9
Scan type:              Progressive scan
Frame rate:             25 fps
Streams:
    #0: Video, H.264 (High Profile level 1.2), 320x180 (DAR 16:9), 25 fps, 4 kb/s
```

So I guess I really have to check against a regex, or check each individual field separately (which might be a better idea).
